### PR TITLE
Kderusso/sparse vector ci failure

### LINKED ISSUE #117904: [CI] SparseVectorQueryBuilderTests testToQuery failing
**Build Scans:**
- [elasticsearch-intake #13866 / part2](https://gradle-enterprise.elastic.co/s/vmf3fvksr6ui2)
- [elasticsearch-intake #13856 / part2](https://gradle-enterprise.elastic.co/s/7tqvkpdfbg2dw)
- [elasticsearch-pull-request #44533 / part-2](https://gradle-enterprise.elastic.co/s/bnbje7ibqwlys)
- [elasticsearch-pull-request #44506 / part-2](https://gradle-enterprise.elastic.co/s/vc64h4cgfc3ya)
- [elasticsearch-periodic-platform-support #5179 / rhel-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/jhfojhlxgtwti)
- [elasticsearch-periodic-platform-support #5179 / oraclelinux-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/6vqfqg52cuori)
- [elasticsearch-periodic-platform-support #5179 / amazonlinux-2_platform-support-aws](https://gradle-enterprise.elastic.co/s/vm5gld3hw2x6o)
- [elasticsearch-periodic-platform-support #5179 / sles-15_platform-support-unix](https://gradle-enterprise.elastic.co/s/rzx3s7z5uldom)
- [elasticsearch-periodic-platform-support #5179 / almalinux-8-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/asn2j4qwvsmhq)
- [elasticsearch-periodic-platform-support #5179 / ubuntu-2004-aarch64_checkpart2_platform-support-arm](https://gradle-enterprise.elastic.co/s/xgccomav5tzoi)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:core:test" --tests "org.elasticsearch.xpack.core.ml.search.SparseVectorQueryBuilderTests.testToQuery" -Dtests.seed=DF0EE2DDB5623279 -Dtests.locale=af-Latn-ZA -Dtests.timezone=Asia/Irkutsk -Druntime.java=23
```

**Applicable branches:**
8.x

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.xpack.core.ml.search.SparseVectorQueryBuilderTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testToQuery),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: null
```

**Issue Reasons:**
- [8.x] 2 consecutive failures in step rhel-9_platform-support-unix
- [8.x] 2 consecutive failures in step oraclelinux-8_platform-support-unix
- [8.x] 2 consecutive failures in step amazonlinux-2_platform-support-aws
- [8.x] 2 consecutive failures in step ubuntu-2004-aarch64_checkpart2_platform-support-arm
- [8.x] 42 failures in test testToQuery (40.8% fail rate in 103 executions)
- [8.x] 6 failures in step part2 (40.0% fail rate in 15 executions)
- [8.x] 10 failures in step part-2 (43.5% fail rate in 23 executions)
- [8.x] 2 failures in step rhel-9_platform-support-unix (100.0% fail rate in 2 executions)
- [8.x] 2 failures in step oraclelinux-8_platform-support-unix (100.0% fail rate in 2 executions)
- [8.x] 2 failures in step amazonlinux-2_platform-support-aws (100.0% fail rate in 2 executions)
- [8.x] 2 failures in step ubuntu-2004-aarch64_checkpart2_platform-support-arm (100.0% fail rate in 2 executions)
- [8.x] 6 failures in pipeline elasticsearch-intake (40.0% fail rate in 15 executions)
- [8.x] 9 failures in pipeline elasticsearch-pull-request (45.0% fail rate in 20 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic (100.0% fail rate in 2 executions)
- [8.x] 2 failures in pipeline elasticsearch-periodic-platform-support (100.0% fail rate in 2 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

### PR DESCRIPTION
**Pull request by kderusso**
Resolves https://github.com/elastic/elasticsearch/issues/117904

### CODE REVIEW

**Review by Samiul-TheSoccerFan — APPROVED** [review]
LGTM

File: x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilderTests.java [hunk 1]
```diff
@@ -232,6 +232,12 @@ public void testToQuery() throws IOException {
 
     private void testDoToQuery(SparseVectorQueryBuilder queryBuilder, SearchExecutionContext context) throws IOException {
         Query query = queryBuilder.doToQuery(context);
+
+        // test query builder can randomly have no vectors, which rewrites to a MatchNoneQuery - nothing more to do in this case.
+        if (query instanceof MatchNoDocsQuery) {
```
Comment (Samiul-TheSoccerFan) [inline x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilderTests.java:237]: If it rewrites to `MatchNoneQuery`, does it mean the test will show as passed without any more validation? Will that be a problem in the future or `SparseVectorQueryWrapper` will never write as `MatchNoneQuery` so there won't be any risk of skipping this test?

Comment (kderusso) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilderTests.java:237]: `SparseVectorQueryBuilder` will rewrite to a `MatchNoneQuery` if `queryVectors` is null. As you can see in the test method, this is definitely something that can happen when we create the test index: 

```
private SparseVectorQueryBuilder createTestQueryBuilder(TokenPruningConfig tokenPruningConfig) {
        SparseVectorQueryBuilder builder;
        if (randomBoolean()) {
            builder = new SparseVectorQueryBuilder(
                SPARSE_VECTOR_FIELD,
                null,
                randomAlphaOfLength(10),
                randomAlphaOfLengthBetween(10, 25),
                tokenPruningConfig != null,
                tokenPruningConfig
            );
        } else {
            builder = new SparseVectorQueryBuilder(
                SPARSE_VECTOR_FIELD,
                WEIGHTED_TOKENS,
                null,
                null,
                tokenPruningConfig != null,
                tokenPruningConfig
            );
        }
```

Comment (Samiul-TheSoccerFan) [reply x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/search/SparseVectorQueryBuilderTests.java:237]: Got it, thank you for the explanation

**Review by jimczi — APPROVED** [review]
Thanks for fixing @kderusso

**Pull request merged**